### DISCUSSION TITLE
[Feature Request] Add a controller / Steam Input API support.

### DISCUSSION THREAD

**Discussion by Redo11** [FakeFishGames/Barotrauma#15449] [Ideas and suggestions]
I am a person who plays Barotrauma on a controller remapped to keyboard and mouse input in Steam.
The game is absolutely playable on a controller, especially with ones with gyro or trackpads (like Steam Controller and Steam Deck).
However, there are certain limitations, like the input precision and inventory management.

Adding support for controller input would be really nice and would fix those problems, as well as allowing for more advanced rebinding.
It would also be nice for a Steam Deck, as I believe controller support is required for the verified rank.
 
Also, use controller support that allows to use both controller and keyboard & mouse inputs simultaneously, as that allows to rebind the controller to player's liking, if preferred.

**Comment by Redo11** [comment]
I understand that controller support is rather a niche issue, due to the game being made specifically for keyboard and mouse, but such implementation would benefit the game in a form of better Steam Deck support, so I hope it is at least looked upon.

**Comment by SomeRandomNoobKekeke** [comment]
You can try this https://steamcommunity.com/sharedfiles/filedetails/?id=2967824521
